server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -666,8 +753,36 @@ private void endSnapshot(SnapshotsInProgress.Entry entry, Metadata metadata) {
             return;
         }
         final Snapshot snapshot = entry.snapshot();
+        if (entry.repositoryStateId() == RepositoryData.UNKNOWN_REPO_GEN) {
```

Comment:
This is actually dead code (became obsolete after improving the master fail-over handling in a recent PR).  `SnapshotsService#processExternalChanges` will remove any such snapshot entry without taking any additional action these days (since 8.x never creates such a snapshot, it can't have listeners for it so quietly dropping it from the cluster state is just fine).